File: webmanager/handlerevents/renamenode/renamenode.js
// Обработчик события (переименование узла (папки))
function renamenode() {
	// При переименовании нужно переименовать объек или систему в БД замечаний (CHOSE и NOTICE)
	var ls = decodeURIComponent(g_fullpath).split(',');
    if (ls.length <= 2) {
		alert('Папку этого уровня переименовать нельзя!');
	    return;
	}
    var newname = prompt('Введите новое название папки', ls[ls.length-1])
	if (newname == null) return;
	if (newname.trim() == '') return;      // Если пробелы, выходим
    $.getJSON('http://' + g_rhost + '/renamenode' + '?path=' + g_fullpath + '&newname=' + encodeURIComponent(newname),
        function(data) {
			if (data == 'notexists') {      // Директория не найдена, кто-то удалил эту папку пока браузер был открыт
			    alert('Переименуемая папка не найдена. Попробуйте обновить дерево!');
				return;
			}
			if (data == 'exists') {         // Папка с таким именем уже существует
			    alert('Папка с таким именем уже существует!');
				return;   
			}
			if (data == 'notrenamed') {          // Ошибка при переименовании папки
			    alert('Не удалось переименовать папку. Возможно нет прав!');
				return;   
			}
   	        if (data == 'success') {
				$('#treeview').jstree('refresh');    // Обновляем дерево
				return;
			}
			console.log(data);
			alert('Сервер вернул неизвестное значение. Операция не выполнена');
		}
    )
}

File: webmanager/handlerevents/renamenode/renamenode.py
import os
import urllib

# Класс обработчик событий (переименование узла)
class RenameNode:
    # Конструктор   
    def __init__(self, app, web):
        app.add_routes([web.get('/renamenode', self.renamenode)])     # Добавление маршрута

    # Обработчик переименования узла дерева
    async def renamenode(self, request):
        params = request.rel_url.query     
        fullpath = self.get_fullpath(params['path'], request.remote)   # Полный путь к выделенной папке

        if not os.path.exists(fullpath):                               # Нет выделенной папки
            return self.web.json_response('notexists')

        newpath = '/'.join(fullpath.split('/')[:-1]) + '/' + urllib.parse.unquote(params['newname'])
        if os.path.exists(newpath):                                    # Папка с таким именем уже существует
            return self.web.json_response('exists')
  
        os.rename(fullpath, newpath)
        if not os.path.exists(newpath):
            return self.web.json_response('notrenamed')                # Не удалось переименовать директорию
        
        return self.web.json_response('success')                       # Сообщаем, что все ОК
